param/parameterized.py

```diff
@@ -3573,39 +3574,65 @@ def _name_if_set(parameterized):
     return '' if default_name else parameterized.name
 
 
-def _get_param_repr(key, val, p, truncate=40):
+def truncate(str_, maxlen = 30):
+    """Return HTML-safe truncated version of given string"""
+    rep = (str_[:(maxlen-2)] + '..') if (len(str_) > (maxlen-2)) else str_
+    return html.escape(rep)
+
+
+def _get_param_repr(key, val, p, vallen=30, doclen=40):
     """HTML representation for a single Parameter object and its value"""
     if hasattr(val, "_repr_html_"):
         try:
             value = val._repr_html_(open=False)
         except:
             value = val._repr_html_()
     else:
-        rep = repr(val)
-        value = (rep[:truncate] + '..') if len(rep) > truncate else rep
+        value = truncate(repr(val), vallen)
 
-    modes = []
-    if p.constant:
-        modes.append('constant')
-    if p.readonly:
-        modes.append('read-only')
-    if getattr(p, 'allow_None', False):
-        modes.append('nullable')
-    mode = ' | '.join(modes)
     if hasattr(p, 'bounds'):
-        bounds = p.bounds
+        if p.bounds is None:
+            range_ = '(-&infin;,&infin;)'
+        elif hasattr(p,'inclusive_bounds'):
+            # Numeric bounds use ( and [ to indicate exclusive and inclusive
+            bl,bu = p.bounds
+            il,iu = p.inclusive_bounds
+            lb = ('[' if il else '(') + ('-&infin;' if bl is None else str(bl))
+            ub = ('&infin;' if bu is None else str(bu)) + (']' if iu else ')')
+            range_ = lb + ', ' + ub
+        else:
+            range_ = repr(p.bounds)
     elif hasattr(p, 'objects') and p.objects:
-        bounds = ', '.join(list(map(repr, p.objects)))
+        range_ = ', '.join(list(map(repr, p.objects)))
+    elif hasattr(p, 'class_'):
+        if isinstance(p.class_, tuple):
```

Comment:
I'm made a change here to handle when `class_` is defined with a tuple of accepted types. I spotted this error with `pn.widgets.FloatSlider.param`.